src/nvim/getchar.c

```diff
@@ -2317,6 +2317,9 @@ static int vgetorpeek(bool advance)
       // try re-mapping.
       for (;;) {
         check_end_reg_executing(advance);
+        if ((mapped_ctrl_c | curbuf->b_mapped_ctrl_c) & get_real_state()) {
```

Comment:
This check is needed because `os_breakcheck()` can call `input_enqueue()`.

Reply:
That would worth a great 1-line comment here.

    // os_breakcheck() can call input_enqueue()